src/components/CenturyFilter.tsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useState } from 'react';
+
+export const CenturyFilter = () => {
+  const [selectCenturies, setSelectCenturies] = useState(new Set());
+
+  useEffect(() => {
+    if (selectCenturies) {
+      setSelectCenturies(selectCenturies);
+    }
+  }, [selectCenturies]);
```

Comment:
The `useEffect` hook here is redundant and potentially problematic. It sets `selectCenturies` to itself whenever it changes, which can cause an infinite loop of updates. Consider removing this `useEffect` unless there's a specific reason for its presence that isn't clear from the current code.